src/Backups/BackupInfo.cpp

```diff
@@ -98,4 +99,23 @@ String BackupInfo::toStringForLogging() const
     return toAST()->formatForLogging();
 }
 
+void BackupInfo::copyS3CredentialsTo(BackupInfo & dest) const
+{
+    /// named_collection case, no need to update
+    if (!dest.id_arg.empty() || !id_arg.empty())
+        throw Exception(ErrorCodes::BAD_ARGUMENTS, "use_same_s3_credentials_for_base_backup is not compatible with named_collections");
+
+    auto & dest_args = dest.args;
+
+    if ((dest_args.size() != 1) && (dest_args.size() != 3))
+        throw Exception(ErrorCodes::LOGICAL_ERROR, "Backup S3 requires 1 or 3 arguments: url, [access_key_id, secret_access_key]");
+
+    // 0 - s3_uri
+    if (dest_args.size() >= 3)
```

Comment:
It can't be `>= 3` because you just checked that it's either `1` or `3`.

Reply:
This is just a copy-paste of the code that you wrote - https://github.com/ClickHouse/ClickHouse/blob/55af6e5c3e1bfdee95845279f33a265a9343dca7/src/Backups/registerBackupEngineS3.cpp#L79-L84

It can be replaced with `size() == 3`, but I don't see any problem with this notation. If you have some preferences, let me know.